src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -168,9 +168,11 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsOperandOfIncrementOrDecrementExpression(SyntaxNode node);
 
         bool IsLeftSideOfDot(SyntaxNode node);
+        bool IsNodeOrAnyAncestorLeftSideOfDot(SyntaxNode node);
         SyntaxNode GetRightSideOfDot(SyntaxNode node);
 
         bool IsRightSideOfQualifiedName(SyntaxNode node);
+        bool IsLeftSideOfExplicitInterfaceSpecifier(SyntaxNode node);
```

Comment:
prefer: IsExplicitInterfaceSpecifier and call that on node.Parent.  There's no real need to know if you'rethe left/right side here (since there isn't even a right side).   If you did want this, it shoudl also be IsNameOfExplicitInterfaceSpecifier as that's what the language syntax model calls this guy.

Reply:
Issue with that approach is the VB implementation uses a regular qualified name for this, so the implementation has to check if given node IsLeftSideOfDot and node.Parent.Parent is ImplementsClauseSyntax